File: JavaScript/problem002.js
exports.sumOfEvenFibs = function() {
    let sum = 0;
    let last = 1;
    let next = 2;
    while (next < 4000000) {
        if (next % 2 == 0) {
            sum += next;
        }
        let temp = next;
        next += last;
        last = temp;
    }
    return sum;
};

File: Java/src/euler/Problem002.java
package euler;

public class Problem002 implements IProblem {

	public String Run() {

		int sum = 0;
		int last = 1;
		int next = 2;

		while (next < 4000000) 
		{
		    if (next % 2 == 0)
		        sum += next;
		  
		    int temp = next;
		    next += last;
		    last = temp;
		}
		
		return String.valueOf(sum);
	}
}
